Python/syntax_test_python.py

```diff
@@ -1490,70 +1490,80 @@ class Starship:
+#                   ^^^^^^ keyword.other.annotation.type-comment.ignore
+#                          ^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.number-sign - comment.line.type-hint
+#                          ^ punctuation.definition.comment
 
 # This is not a type-ignore comment.
 value = 3  # type: ignore_class_starting_with_ignore
-#          ^ comment.line.type-hint.python punctuation.definition.comment.python comment.line.number-sign.python
-#            ^^^^ comment.line.type-hint.python keyword.other.annotation.type-comment.python
-#                ^ comment.line.type-hint.python punctuation.separator.annotation.type-comment.python
-#                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.type-hint.python support.class.python
+#          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.type-hint - comment.line.number-sign
+#          ^ punctuation.definition.comment
+#            ^^^^ keyword.other.annotation.type-comment
+#                ^ punctuation.separator.annotation.type-comment
+#                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ support.class
 
 # Mypy ignore extension
-value = 3  # type: ignore[error-code-1, 4noint][error]
-#                  ^^^^^^ comment.line.type-hint.python keyword.other.annotation.type-comment.ignore
-#                        ^ comment.line.type-hint.python comment.line.type-hint.python punctuation.section.brackets.begin.python
-#                         ^^^^^^^^^^^^ comment.line.type-hint.python comment.line.type-hint.python.mypy-errorcode
-#                                     ^ comment.line.type-hint.python punctuation.separator.sequence.python.mypy-errorcode
-#                                       ^ invalid.illegal.python
-#                                             ^ comment.line.type-hint.python punctuation.section.brackets.end.python
-#                                              ^ invalid.illegal.python
+value = 3  # type: ignore[error-code-1, 4noint][error] # foo
+#          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.type-hint - comment.line.number-sign
+#                  ^^^^^^ keyword.other.annotation.type-comment.ignore
+#                        ^ punctuation.section.brackets.begin
+#                         ^^^^^^^^^^^^ constant.other.error-code
+#                                     ^ punctuation.separator.sequence
+#                                       ^^^^^^ invalid.illegal
+#                                             ^ punctuation.section.brackets.end
+#                                              ^^^^^^^ invalid.illegal
+#                                                      ^^^^^ comment.line.number-sign - comment.line.type-hint
 
 # With/for allow commas
 for a, b in lst: # type: str, int
-#                           ^ comment.line.type-hint.python punctuation.separator.sequence.python
+#                           ^ comment.line.type-hint punctuation.separator.sequence
 
 
 primes = 5  # type: List[Dict[*property:str, ...]], bool # comment
```

Comment:
```
primes = 5  # type: List[Dict[*property:str, ...]], bool # comment
                                                  ^^^^^^ invalid
```

This looks like a tuple but is invalid.  E.g. `# type: int, str` is invalid.

Reply:
Is this valid syntax? I haven't seen this mentioned anywhere: `Dict[property:str, ...]`

Reply:
`Dict[foo:bar, ...]` is not valid; neither the `:` nor the `...` in that position.